File: FreeLine-Android/app/src/main/java/com/freeline/app/numbers/NumberModels.kt
package com.freeline.app.numbers

data class AvailableNumberOption(
    val phoneNumber: String,
    val nationalFormat: String,
    val locality: String,
    val region: String,
    val provider: String,
) {
    val areaCode: String
        get() = phoneNumber.removePrefix("+1").take(3)
}

data class AssignedNumber(
    val assignmentId: String,
    val assignedAt: String,
    val activationDeadline: String,
    val areaCode: String,
    val externalId: String,
    val locality: String,
    val nationalFormat: String,
    val phoneNumber: String,
    val phoneNumberId: String,
    val provider: String,
    val quarantineUntil: String?,
    val region: String,
    val releasedAt: String?,
    val status: String,
    val userId: String,
)

File: FreeLine-iOS/Sources/Numbers/NumberModels.swift
import Foundation

struct AvailableNumberOption: Codable, Equatable {
    let phoneNumber: String
    let nationalFormat: String
    let locality: String
    let region: String
    let provider: String

    var areaCode: String {
        String(phoneNumber.dropFirst(2).prefix(3))
    }
}

struct AssignedNumber: Codable, Equatable {
    let assignmentId: String
    let assignedAt: String
    let activationDeadline: String
    let areaCode: String
    let externalId: String
    let locality: String
    let nationalFormat: String
    let phoneNumber: String
    let phoneNumberId: String
    let provider: String
    let quarantineUntil: String?
    let region: String
    let releasedAt: String?
    let status: String
    let userId: String
}
